File: FinanceBackEnd/Finance.Domain/Models/Subscriptions/Subscription.cs
using Finance.Domain.Enums;
using Finance.Domain.Models.Base;
using Finance.Domain.Models.Currencies;
using Finance.Domain.SpecialTypes;

namespace Finance.Domain.Models.Subscriptions;

public class Subscription() : AuditedEntity<Guid>()
{
    public Guid CurrencyId { get; set; }
    public Currency Currency { get; set; } = default!;
    public string Name { get; set; } = string.Empty;
    public Money Price { get; set; } = 0;
    public FrequencyEnum Frequency { get; set; } = FrequencyEnum.Monthly;
}

File: FinanceFrontEnd/FinanceApp/app/types/subscription.ts
import type { CatalogItem } from '@/types/catalog';

export enum SubscriptionFrequency {
  Monthly = 'monthly',
  Annual = 'annual',
}

export type SubscriptionFrequencyName = 'monthly' | 'annual';

export interface Currency {
  id: string;
  name: string;
  shortName: string;
}

export interface Subscription {
  id: string;
  name: string;
  price: number;
  frequency: SubscriptionFrequency | SubscriptionFrequencyName;
  currencyId: string;
  currency?: Currency;
  paymentDate?: string | null;
}

export interface SubscriptionsData {
  subscriptions: {
    items: Subscription[];
    totalCount: number;
  };
  currencies: CatalogItem[];
  frequencies: Array<{
    id: number;
    name: string;
  }>;
}
